public/app/plugins/panel/geomap/layers/data/markersLayer.tsx

```diff
@@ -131,7 +135,22 @@ export const markersLayer: MapLayerRegistryItem<MarkersConfig> = {
 
         // Source reads the data and provides a set of features to visualize
         const vectorSource = new source.Vector({ features });
-        vectorLayer.setSource(vectorSource);
+        if (config?.cluster?.enabled) {
+
+          const clusterSource = new source.Cluster({
+            distance: config.cluster?.distance ?? defaultClusterConfig.distance,
+            minDistance: config.cluster?.minDistance ?? defaultClusterConfig.minDistance,
+            source: vectorSource,
+          });
+          vectorLayer.setSource(clusterSource);
+          const origStyle = vectorLayer.getStyle();
+          if (origStyle && 'getImage' in origStyle) {
+            vectorLayer.setStyle(getClusterStyle(origStyle, style));
```

Comment:
That's what I started with, too, but with the current cluster config, all of the distance stuff is handled with the cluster source, which handles grouping the features together. 

Originally, I thought it would have to be a unique style maker specific to clustering, but I'm hoping that this is a way that we can use the same style makers as in a non-clustered layer. 

Currently, trying to chew on how to "summarize" a cluster. I'm hoping there are some clean options for scaling and potentially colors. However, it will need to be more nuanced if we add custom icons and potentially click behavior.